src/backend/distributed/shared_library_init.c

```diff
@@ -2429,6 +2465,41 @@ RegisterCitusConfigVariables(void)
 		GUC_NO_SHOW_ALL | GUC_NOT_IN_SAMPLE,
 		NULL, NULL, NULL);
 
+	DefineCustomIntVariable(
+		"citus.stat_counters_flush_interval",
+		gettext_noop("Sets the interval to flush the local stat counters into "
+					 "the shared memory."),
+		gettext_noop("Stat counters are used to track the number of certain "
+					 "operations in Citus. While setting this GUC to -1 disables "
+					 "stat counters, setting it to a positive value will flush "
+					 "the local stat counters into the shared memory every "
+					 "interval milliseconds instead of flushing them immediately "
+					 "after the operation, as it is done when the value is 0. "
+					 "Higher values reduce the overhead of flushing the stat "
+					 "counters but increase the time it takes to see the updated "
+					 "stat counters."),
+		&StatCountersFlushInterval,
+		DEFAULT_STAT_COUNTERS_FLUSH_INTERVAL,
+		DISABLE_STAT_COUNTERS_FLUSH_INTERVAL,
+		5 * MS_PER_MINUTE,
+		PGC_POSTMASTER,
```

Comment:
```suggestion
		PGC_POSTMASTER | PGC_SUSET,
```

Reply:
Also reconsider if we really want to enforce PGC_POSTMASTER vs if we're okay with allocating the shared hash table even if the feature is disabled.